lib/pf/vlan.pm

```diff
@@ -108,7 +108,7 @@ sub fetchVlanForNode {
 
     # there were no violation, now onto registration handling
     ($registration,$role) = $this->getRegistrationVlan($switch, $ifIndex, $mac, $node_info, $connection_type, $user_name, $ssid, $radius_request, $realm, $stripped_user_name, $autoreg);
-    if (defined($registration) && $registration != 0) {
+    if (defined($registration) && $registration ne 0) {
```

Comment:
I would change it to $registration ne "0" so it is clearer that we are comparing a string.